crates/ty_python_semantic/resources/mdtest/generics/legacy/functions.md

```diff
@@ -455,3 +455,43 @@ def overloaded_outer(t: T | None = None) -> None:
     if t is not None:
         inner(t)
 ```
+
+## Unpacking a TypeVar
+
+We can infer precise heterogeneous types from the result of an unpacking operation applied to a
+TypeVar if the TypeVar's upper bound is a type with a precise tuple spec:
+
+```py
+from dataclasses import dataclass
+from typing import NamedTuple, Final, TypeVar, Generic
+
+T = TypeVar("T", bound=tuple[int, str])
+
+def f(x: T) -> None:
+    a, b = x
+    reveal_type(a)  # revealed: int
+    reveal_type(b)  # revealed: str
```

Comment:
As written, this is a useless use of a typevar (since it appears only once in the signature), which we might at some point want to emit a diagnostic on? This annotation should just be replaced with a direct `tuple[int, str]` instead.

I think we should either have a TODO comment about that, or (maybe less distracting) change this signature so the function e.g. also returns a `T`.

```suggestion
def f(x: T) -> T:
    a, b = x
    reveal_type(a)  # revealed: int
    reveal_type(b)  # revealed: str
    return x
```